required/latex-lab/latex-lab-block.dtx

```diff
@@ -397,1262 +547,3129 @@
+%    on which the instance name differs. Again we pass it the optional
+%    key/value list.
+%    \begin{macrocode}
+  \tl_if_empty:NF \l_@@_inner_instance_tl
+    {
+%    \end{macrocode}
+%    We expand the first two arguments so that we get proper names for
+%    template type and instance, because \cs{UseInstance} is not doing
+%    that for us in the right way.
+%  \changes{v0.9j}{2025/07/11}{Expand arguments before passing them to
+%    \cs{UseInstance} to get better error messages}
+%    \begin{macrocode}
+      \exp_args:Nee
+      \UseInstance{ \l_@@_inner_instance_type_tl }
+                  { \l_@@_inner_instance_tl
+                    \tl_if_empty:NF \l_@@_inner_level_counter_tl
+                       %   not clean use "o"?
+                       { - \int_use:N \l_@@_inner_level_counter_tl }
+                  }
+                  \l_@@_unused_blockenv_keys_tl
+                    #2      % <-- \BooleanTrue or False
+                  { #3 }    % <-- \NoValue or content
+                  { #4 }    % <-- \NoValue or content
+%    \end{macrocode}
+%    Again the instance may have processed a few keys from the so far
+%    unused keys, so we update \cs{l_@@_unused_blockenv_keys_tl} to
+%    match the new reality.
+%    \begin{macrocode}
+      \tl_set_eq:NN \l_@@_unused_blockenv_keys_tl \UnusedTemplateKeys
+    }
+%    \end{macrocode}
+%    At this point, the \cs{l_@@_unused_blockenv_keys_tl} token list
+%    should either be empty or it should contain only keys that are suitable for
+%    the item template, but right now there is no code to test that
+%    can test
+%    the latter; it would help probably if we have an interface for
+%    this.
+%
+%    For now we handle that when the first item is encountered, but
+%    that isn't really clean.\fmi{fix}
+%    \begin{macrocode}
+%  \tl_if_empty:NF \l_@@_unused_blockenv_keys_tl
+%    {
+%       % check if only item template keys remain
+%    }
+%    \end{macrocode}
+%    If this is supposed to be a transparent block environment then we
+%    have to decrement the nesting level again so that nested
+%    environments think nothing is there.
+%    \begin{macrocode}
+  \bool_if:NT \l_@@_transparent_level_bool
+     {  \int_gdecr:N \g_block_nesting_depth_int }
+%    \end{macrocode}
+%    We finish off with \cs{l_@@_final_code_tl} which defaults to
+%    \tn{ignorespaces} so that spaces between \verb=\begin{...}= and
+%    the start of the text are ignored.
+%    \begin{macrocode}
+  \l_@@_final_code_tl
+}
+%    \end{macrocode}
+%  \end{template}
+%
+%
+%
+%  \begin{macro}{\BlockEnv,\SimpleBlockEnv}
+%    To simplify the environment declarations later we provide two
+%    simply commands that invoke a \insttype{blockenv} instance. The
+%    matching counterpart to these commands is \cs{BockEnvEnd}
+%    (defined below) that carries out all necessary action when a
+%    block environment ends.
+%    \begin{macrocode}
+\cs_new_protected:Npn \BlockEnv            % #1#2#3#4 implicit
```

Comment:
In general for doc commands we keep the args explicit - I think here it's OK but we should just be clear on that.